CMakeLists.txt

```diff
@@ -1,359 +1,205 @@
-cmake_minimum_required (VERSION 3.2)
-project(xgboost)
+cmake_minimum_required(VERSION 3.2)
+project(xgboost LANGUAGES CXX C VERSION 0.82)
```

Comment:
Any reason why we put the version number here?

Reply:
@hcho3 It will be exported to the CMake install file, used by client CMake script `find_package(xgboost 0.82)`

Reply:
@trivialfis Would a user want to install multiple versions of XGBoost? Otherwise, the version of XGBoost is determined by the git commit of cloned XGBoost repository.

Reply:
@hcho3 
> Would a user want to install multiple versions of XGBoost

I don't know.  But it seems weird that there's no way to determine the version of a package?  When you try to find your dependency, making sure it's greater than some version seems to be a common practice right?  Like using `find_package(Boost 1.64)` to make sure boost is **at least** 1.64 ...

It's possible to call `git log` then process the strings to get the correct version.  But let leave it as the last resort.

Reply:
@trivialfis Ah so the user can specify the minimum required version of XGBoost. Got it.